crates/stages/stages/src/sets.rs

```diff
@@ -65,6 +65,7 @@ use tokio::sync::watch;
 /// - [`BodyStage`]
 /// - [`SenderRecoveryStage`]
 /// - [`ExecutionStage`]
+/// - [`PruneStage`] (execute)
 /// - [`MerkleStage`] (unwind)
 /// - [`AccountHashingStage`]
 /// - [`StorageHashingStage`]
```

Comment:
the reason why I put this stage here is because otherwise we'll prune senders, it will create a huge DB freelist and it won't be consumed by other DB insertions

having Prune stage right after Execution should be fine:
1. The freelist created from pruning senders will be filled with new data in consequent stages
2. Account and Storage history will be pruned on the next pipeline run, and it's guaranteed for us to have two of them on initial sync: one to sync from genesis to N, and the other to sync new blocks from N that accumulated during the first sync

Reply:
nice. do we still want an issue for a stage for initial prune then for history before node gives green light to start validating? what are the initial pruning times for history?

Reply:
hmm, I thought more and agreed that it should be last, there's another reason for that: hashing stages require changesets on **non-first** run of the pipeline https://github.com/paradigmxyz/reth/blob/207aeeb5caef38cfa208101540354f60201104d7/crates/stages/stages/src/stages/execution.rs#L143-L170

it means that if we run the Prune stage after Execution, we will delete changesets that will be required for hashing stages

moved Prune to the end, before Finish. Not ideal that we will prune senders and free 150GB of data that won't be immediately occupied by other data, though.

Reply:
hmm, maybe we can have a stage that prunes just the senders after execution, and another stage that runs pruner for all segments. Not perfect because we'll have two prune stages, but it solves our problem

Reply:
added a separate `PruneSenderRecoveryStage` that uses the `PruneStage` under the hood, @mattsse fyi https://github.com/paradigmxyz/reth/pull/9419/commits/4c42a75927c9d2e52e4c00c85d835049e2c7e24b